tests/test_widgets.py

```diff
@@ -1129,6 +1129,135 @@ def test_multi_column_list_box(self):
         ]
         self.assertEqual(mc_list.value, 0)
 
+
+    def test_multi_column_list_box_delimiter(self):
+        """
+        Check MultiColumnListBox works as expected with space_delimiter
+        """
+        # Create a dummy screen.
+        screen = MagicMock(spec=Screen, colours=8, unicode_aware=False)
+        scene = MagicMock(spec=Scene)
+        canvas = Canvas(screen, 10, 40, 0, 0)
+
+        # Create the form we want to test.
+        form = Frame(canvas, canvas.height, canvas.width, has_border=False)
+        layout = Layout([100], fill_frame=True)
+        mc_list = MultiColumnListBox(
+            Widget.FILL_FRAME,
+            [3, "4", ">4", "<4", "^10%", "100%"],
+            [
+                (["1", "2", "3", "4", "5", "6"], 1),
+                (["11", "222", "333", "444", "555", "6"], 2),
+                (["111", "2", "3", "4", "5", "6"], 3),
+                (["1", "2", "33333", "4", "5", "6"], 4),
+                (["1", "2", "3", "4", "5", "6666666666666666666666"], 5),
+            ],
+            titles=["A", "B", "C", "D", "E", "F"],
+            name="mc_list",
+            space_delimiter='|')
+        form.add_layout(layout)
+        layout.add_widget(mc_list)
+        form.fix()
+        form.register_scene(scene)
+        form.reset()
+
+        # Check we have a default value for our list.
```

Comment:
This clearly tests a lot more than just the new delimiter... Can you please remove all the other tests and just leave the stuff for your new function?

Reply:
Done.